### ISSUE TITLE
[Enhancement]: Circle Painter - Enhancement (draft)

### ISSUE THREAD

**Issue by claws99** [LinwoodDev/Butterfly#476]
### Is your feature request related to a problem? Please describe

What is currently being called `Circle` is technically an ellipse and its is extremely difficult to get a circle.

providing another painter for circle isnt desirable.

Drawing circle can became easy if we provide any method to constrain proportions (width = height)

### Describe your feature request!

I'm thinking we can provide something like holding something else  which indicate drawing.

Using keyboard:
(borrowing from Inkscape)
- draw = ellipse
- `shift` + draw = draw the ellipse with starting point as center
- `ctrl` + draw = circle
- `ctrl + shift` + draw = draw circle  with starting point as center

Using touch:
?? I'll research a bit more.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by CodeDoctorDE** [comment]
There i already a circle painter. You can find it under the shapes

**Comment by claws99** [comment]
> There i already a circle painter. You can find it under the shapes

yeah. I'm using it, its an enhancement proposal for the current circle painter.

**Comment by CodeDoctorDE** [comment]
Hmm, the problem is: The aspect ratio is a property of the shape painter.
We can override this behavior when pressing shift or ctrl if you want.
I would like to have a similar behavior on mobile. It would be cool if there is a similar behavior on mobile devices

**Comment by claws99** [comment]
>I would like to have a similar behavior on mobile. It would be cool if there is a similar behavior on mobile devices

I'm with you on providing similar behavior. Providing similar behaviour is what makes app coherent.  

Just to clarify my thinking, similar behavior shouldn't mean identical behaviour. As identical features in most of the cases won't be the optimal/best possible solutions on all device types. This is fundamentally because different devices have different input  and output devices.

This just simply means:
- For devices with touch input, optimizing app to be easily interactable using gestures 
- For devices with keyboard input,  optimizing app to be easily interactable using keyboard keys.
- etc

This is inline with #433

Example: Excalidraw is great to be used with mouse, but its a nightmare on smaller and touch&pen-input devices.

In short, the principle I'm referring to is ["There's more than one way to do it," commonly known as TMTOWTDI, ](https://en.wikipedia.org/wiki/Perl#Overview)

**Comment by CodeDoctorDE** [comment]
The circle behavior can be replicated when setting aspect ratio to one. But i don't think it's really user friendly on mobile

**Comment by claws99** [comment]
Sure! your call :)

**Comment by CodeDoctorDE** [comment]
Added in 2.0.0-beta.6
Recoded the shape painter.
The same system can be used in a new stamp painter (#stamp
